src/serializer/functions.js

```diff
@@ -126,11 +127,27 @@ export class Functions {
     return recordedAdditionalFunctions;
   }
 
+  // This will also handle postprocessing for PossiblyNormalCompletion
   _createAdditionalEffects(effects: Effects): AdditionalFunctionEffects {
-    return {
-      effects,
-      transforms: [],
-    };
+    let [result, generator] = effects;
+    let retValue: AdditionalFunctionEffects = { effects, transforms: [] };
+    // Create the effects, arguments and buildNode for the return value, saving them in AdditionalFunctionEffects
+    if (result instanceof PossiblyNormalCompletion) {
+      let { joinCondition, consequent, alternate, consequentEffects, alternateEffects } = result;
+      let joinedEffects = Join.joinEffects(this.realm, joinCondition, consequentEffects, alternateEffects, true);
```

Comment:
I don't understand why you are doing the join at this point. It does not seem like a natural join point to me. In particular, exceptions does not get swallowed at this point, so they should not be joining their effects with the normal effects here. Let's discuss this face to face.